From 76bbc0155e404bbe0f5475a3a30197f7efe3e603 Mon Sep 17 00:00:00 2001
From: neomilium <neomilium@gmail.com>
Date: Sat, 13 Mar 2010 00:40:58 +0100
Subject: [PATCH 6/8] Fix destination filename extension
  - In case of HD quality, the extension is mp4 (not .flv) so extension is now detected using URL

---
 canal.py |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/canal.py b/canal.py
index 4b73f99..3b7657f 100755
--- a/canal.py
+++ b/canal.py
@@ -149,13 +149,13 @@ def downloadURL(url, verbose, overwrite, force, rename):
             with --directory option.")
 
     #Generate the related command
-    file_address = ""
+    filename = ""
     if rename:
-        file_address = join(dest_directory, video_rubrique, video_rubrique
-        +"_"+video_date.replace('/', '_')+".flv")
+        fileext = url.split('.').pop()
+        filename = video_rubrique + "_" + video_date + "." + fileext
     else:
-        file_address = join(dest_directory, video_rubrique,
-        url.split('/').pop())
+        filename = url.split('/').pop()
+    destination = join(dest_directory, video_rubrique, filename)
 
     cmd ="flvstreamer -eqr "+url+" -o "+file_address
     if verbose:
-- 
1.6.3.3

